pkg/binding/bindingpolicy.go

```diff
@@ -101,6 +118,73 @@ func (c *Controller) syncBindingPolicy(ctx context.Context, bindingPolicyName st
 	return c.deleteResolutionForBindingPolicy(ctx, bindingPolicyName)
 }
 
+func (c *Controller) createOrUpdateStatusCollectorAvailableCondition(ctx context.Context, bp *v1alpha1.BindingPolicy, missingSCs []string) error {
+	// compose tentative condition where LastTransitionTime is TBD
+	conditionTentative := v1alpha1.BindingPolicyCondition{}
+	if len(missingSCs) != 0 {
+		conditionTentative = v1alpha1.BindingPolicyCondition{
+			Type:    v1alpha1.TypeStatusCollectorsAvailable,
+			Status:  corev1.ConditionFalse,
+			Reason:  v1alpha1.ReasonReconcileError,
+			Message: fmt.Sprintf("Missing StatusCollector(s) %s", missingSCs),
+		}
+	} else {
+		conditionTentative = v1alpha1.BindingPolicyCondition{
+			Type:    v1alpha1.TypeStatusCollectorsAvailable,
+			Status:  corev1.ConditionTrue,
+			Reason:  v1alpha1.ReasonReconcileSuccess,
+			Message: "All StatusCollector(s) are available",
+		}
+	}
+	// check whether bp already has a condition that is the same (excluding LastTransitionTime) as the tentative one
+	hasSame := false
+	for _, conditionActual := range bp.Status.Conditions {
+		if v1alpha1.AreConditionsEqual(conditionActual, conditionTentative) {
+			hasSame = true
+			break
+		}
+	}
+	// create or update if necessary
+	if hasSame {
+		return nil
+	}
```

Comment:
Wouldn't it be easier to have v1alpha1.SetCondition do the equality test (and return the result) and update the time if it is a change?

Reply:
Improved.